art_tree.go

```diff
@@ -70,30 +70,29 @@ func (t *ArtTree) searchHelper(current *ArtNode, key []byte, depth int) *ArtNode
 	// While we have nodes to search
 	if current != nil {
 		maxKeyIndex := len(key) - 1
+		if depth > maxKeyIndex {
+			return current
+		}
 
-		// Check if the current is a match
-		if current.IsLeaf() {
-			if current.IsMatch(key) {
-				return current
-			}
-
-			// Bail if no match
-			return nil
-		} else if depth > maxKeyIndex {
+		// Check if the current is a match (including prefix match)
+		if current.IsLeaf() && len(current.key) >= len(key) && bytes.Equal(key, current.key[0:len(key)]) {
```

Comment:
right, i can see you're adjusting the tests so that we trap scenarios where we have partial matches.  i think that's primarily where the `prefixMismatch` function is flawed.  I think, though, that the fix should probably move upstream.  It appears that `prefixMismatch` currently uses a `0` to represent both an invalid match, and a match on the first character....  so, I believe the fix should go there... will take some thought.